python_modules/dagster/dagster/_core/pipes/context.py

```diff
@@ -108,12 +119,31 @@ def _resolve_metadata_value(
 
     # Type ignores because we currently validate in individual handlers
     def handle_message(self, message: PipesMessage) -> None:
-        if message["method"] == "report_asset_materialization":
+        if self._received_closed_msg:
+            # should we be more flexible incase complexity causes some other messages to be flushed after close?
+            raise DagsterPipesExecutionError(
+                f"Received message after pipes session closed: `{message}`"
+            )
```

Comment:
seems like we should warn for now until we understand this a little more. Seems like threading bugs could trigger this